synapse/handlers/sync.py

```diff
@@ -2121,7 +2123,17 @@ async def _get_room_changes_for_incremental_sync(
             room_entry = room_to_events.get(room_id, None)
 
             newly_joined = room_id in newly_joined_rooms
-            if room_entry:
+            if room_id in sync_result_builder.retransmit_room_ids:
+                entry = RoomSyncResultBuilder(
+                    room_id=room_id,
+                    rtype="joined",
+                    events=None,
+                    newly_joined=True,
+                    full_state=True,
+                    since_token=None,
+                    upto_token=now_token,
```

Comment:
MV: are these parameters correct? They are different below.
DMR: wtf is going on?

Reply:
since_token=None is correct. There's a lot of overlap between this and newly_joined=True full_state=True; the params aren't orthogonal

more sum types pls